source/common/http/http2/codec_impl.cc

```diff
@@ -1426,6 +1429,69 @@ void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
   }
 }
 
+void ConnectionImpl::dumpStreams(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+
+  // Try to dump details for the current stream.
+  // If none, dump a subset of our active streams.
+  os << spaces << "current_stream_id_: ";
+  if (current_stream_id_.has_value()) {
+    os << current_stream_id_.value() << ",";
+  } else {
+    os << "null,";
+  }
+  os << " Number of active streams: " << active_streams_.size();
+
+  if (current_stream_id_.has_value()) {
+    os << " Dumping current stream:\n";
+    for (auto& stream : active_streams_) {
+      if (stream->stream_id_ == current_stream_id_.value()) {
+        DUMP_DETAILS(stream);
+        break;
+      }
+    }
+  } else {
+    const auto streams_to_dump = std::min<size_t>(active_streams_.size(), 25);
+    os << " Dumping " << streams_to_dump << " Active Streams:\n";
+    std::for_each_n(active_streams_.begin(), streams_to_dump,
+                    [&](auto& stream) { DUMP_DETAILS(stream); });
+  }
+}
+
+void ClientConnectionImpl::dumpStreams(std::ostream& os, int indent_level) const {
+  ConnectionImpl::dumpStreams(os, indent_level);
+
+  if (!current_stream_id_.has_value()) {
+    return;
+  }
+
+  // Try to dump the downstream request information, corresponding to the
+  // stream we were processing.
+  const char* spaces = spacesForLevel(indent_level);
+  os << spaces << "Dumping corresponding downstream request for upstream stream "
+     << current_stream_id_.value() << ":\n";
+
+  for (auto& stream : active_streams_) {
```

Comment:
Consider ConnectionImpl::getStream instead.

I can see iterating over active_streams_ being lower dependency and thus preferable when running from a crash handler.

Reply:
I would use getStream.

Reply:
Done.